src/util/fileUtils.js

```diff
@@ -0,0 +1,50 @@
+// Maximum file upload size in bytes (50 MB)
+export const MAX_FILE_SIZE_BYTES = 50 * 1024 * 1024; // 52,428,800 bytes
+export const MAX_FILE_SIZE_MB = 50;
+
+/**
+ * Validates if a file size is within the allowed limit
+ * @param {number} sizeInBytes - File size in bytes
+ * @returns {boolean} True if valid, false otherwise
+ */
+export const isFileSizeValid = (sizeInBytes) => {
+  return sizeInBytes <= MAX_FILE_SIZE_BYTES;
+};
+
+/**
+ * Formats bytes to human-readable format
+ * @param {number} bytes - Size in bytes
+ * @param {number} decimals - Number of decimal places (default: 2)
+ * @returns {string} Formatted size string (e.g., "45.5 MB")
+ */
+export const formatBytes = (bytes, decimals = 2) => {
+  if (bytes === 0) return '0 Bytes';
+
+  const k = 1024;
+  const dm = Math.max(0, decimals);
+  const sizes = ['Bytes', 'KB', 'MB', 'GB'];
+
+  const i = Math.floor(Math.log(bytes) / Math.log(k));
+
+  const value = Number.parseFloat((bytes / (k ** i)).toFixed(dm));
+
+  return `${value} ${sizes[i]}`;
```

Comment:
Consider improving naming: `k`, `dm`, `i` are not intuitive. Would be better to use descriptive terms.

```suggestion
  const base = 1024;
  const decimalPlaces = Math.max(0, decimals);
  const units = ['Bytes', 'KB', 'MB', 'GB', 'TB', 'PB'];

  const unitIndex = Math.floor(Math.log(bytes) / Math.log(base));
  
  const value = Number((bytes / base ** unitIndex).toFixed(decimalPlaces));

  return `${value} ${units[unitIndex]}`;
  
```